### DISCUSSION TITLE
Verifying build output, using Verify without testing framework

### DISCUSSION THREAD

**Discussion by ap0llo** [VerifyTests/Verify#800] [Ideas]
Hi,

I've been using Verify for a few projects and it always worked like a charm, great work!

One scenario where I'm unsure about what the best approach would be is verifying files as part of the build process of my project (e.g. verifying the contents of the NuGet package being built).
This does not really fit well into the "write a test" approach because a test typically does not have access to the packages in the build output.

For that purpose I'm thinking of creating a console application (or a [Cake Addin](https://cakebuild.net/)) that I can call to verify a file.

I'd be interested if anyone as already done something like this.
Also - since I'm unsure if Verify can even be outside of the context of a testing framework like xunit or NUnit and is even suited for my use case - I'd be interested in your opinions if this can be achieved and be thankful for any pointers of how this could look like.

An alternative I'm considering is implementing the serialization & comparison directly in the console application and only reusing e.g. the DiffEngine package.

Thanks in advance!

**Comment by SimonCropp** [comment]
> This does not really fit well into writing a test for this because a test typically does not have access to the packages in the build output.

why? a nuget can be produced by the build. and a test can open that up and verify it

**Reply by ap0llo** [reply]
I may be overthinking this, but I would like to keep knowledge about the build output structure (e.g. where do packages go) within the build logic and thus avoid making assumption about it in the test code.
